models/cms_hcc/final/cms_hcc__patient_risk_scores.sql

```diff
@@ -24,6 +24,18 @@ with seed_adjustment_rates as (
 
 )
 
+, member_months as (
+
+    select
+        patient_id
+        , cast(substr(year_month, 1, 4) as integer) as eligible_year
+        , COUNT(1) as member_months
+    from  {{ ref('financial_pmpm__member_months') }}
+    group by
```

Comment:
Can you add a payment_year filter to ensure that the member months being used in this calculation are from the correct date range? You can see how we do this in the [intermediate model](https://github.com/tuva-health/tuva/blob/b2c7435a34b58b8c6098efbbce9eba82d443e4bd/models/cms_hcc/intermediate/cms_hcc__int_members.sql#L39) for eligibility. You can include the Jinja code for payment year at the top and then add a where clause to make sure your eligible_year matches the payment_year.

Reply:
@sarah-tuva It technically isn't necessary as in the join I am joining on year, thus it wouldn't matter. I can include for performance considerations if it is necessary. The larger question I have is: Is tuva constrained to run a single payment year at a time? Or could multiple payment years run at once? If it is the latter or the latter is desirable in a future state, then I would opt to not have the filter. If it is only designed to run serially one year at a time, then I can add the filter.

Reply:
@pfehlinger I missed that join down below. That works for me! 

To answer your question, the data mart is currently designed to run one payment year at a time. Snapshots can be enabled to support analytics over multiple payment years. We have had some internal discussions around refactoring this mart to run all possible payment years with the given data. This would change the ability for someone to run analytics on any year they choose, though. I would love to hear your feedback on what you think would be most useful.